### PR TITLE
Fix integration with Synapse

### LINKED ISSUE #9: Prompt for matrix homeserver URL
Hello, 

I think we should be prompted for the homeserver URL when installing. 
I installed this, and the URL which it tries to connect me on webui is not the one of my homeserver, but the synpase-admin URL. 

What do you think?

### PR DESCRIPTION
**Pull request by tituspijean**
Closes #9 

`/_synapse/admin` endpoint needs to be added in the NGINX config of Synapse's domain.

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by nathanael-h** [comment]
I tested this branch. If I install on the same domain as synapase, the nginx.conf is overwritten by the synapse.nginx.conf because they both end is the same dir, with the same name. 
Full logs : https://paste.yunohost.org/raw/iwovavemux

**Comment by tituspijean** [comment]
>  If I install on the same domain as synapase

What. It is supposed to be on its own domain. How could your YunoHost let you do that?!

**Comment by nathanael-h** [comment]
Mmmh I was surprised too. I tried on the same domain because of CORS issues. Will try on dedicated domain :+1:

**Comment by tituspijean** [comment]
You may try again, with the commit above, but I am still flabbergasted you could install it on the same domain as Synapse.

**Comment by tituspijean** [comment]
!testme

**Comment by nathanael-h** [comment]
Cool! I'll test this new commit on same domain. 
On a dedicated domain I still have the CORS error. 
image

By the way I tested with curl, and I have the SSO. and not the json response.. : 
```
nath@aaaaaaa:~$ curl -I -L --header "Authorization: Bearer syt_xxxxxxxxxxxxxxxxxxxxxxx" https:/matrix.isidorus.fr/_synapse/admin/v1/server_version
HTTP/2 302 
server: nginx
date: Tue, 01 Mar 2022 13:10:36 GMT
content-type: text/html
content-length: 154
location: https://domain.fr/yunohost/sso/?r=aHR0cHM6Ly9tYXRyaxxxxxxxxxxxxYWRtaW4vdjEvc2VydmVyX3ZlcnNpb24=
x-sso-wat: You've just been SSOed
content-security-policy: upgrade-insecure-requests
content-security-policy-report-only: default-src https: data: blob: ; object-src https: data: 'unsafe-inline'; style-src https: data: 'unsafe-inline' ; script-src https: data: 'unsafe-inline' 'unsafe-eval'
x-content-type-options: nosniff
x-xss-protection: 1; mode=block
x-download-options: noopen
x-permitted-cross-domain-policies: none
x-frame-options: SAMEORIGIN
permissions-policy: interest-cohort=()
strict-transport-security: max-age=63072000; includeSubDomains; preload
``

File: scripts/remove [hunk 1]
```diff
@@ -70,9 +72,12 @@ ynh_secure_remove --file="$final_path"
 #=================================================
 ynh_script_progression --message="Removing NGINX web server configuration..." --weight=5
 
-# Remove the dedicated nginx config
+# Remove the dedicated NGINX config
 ynh_remove_nginx_config
 
+# Remove NGINX config that enabled /_synapse/admin endpoint
+ynh_secure_remove --file="/etc/nginx/conf.d/${synapse_domain}.d/synapse-admin.conf"
+
```
Comment (nathanael-h) [inline scripts/remove:80]:
```suggestion
ynh_secure_remove --file="/etc/nginx/conf.d/${synapse_domain}.d/$app.endpoint.conf"

```
This file was not removed.

**Comment by nathanael-h** [comment]
Different tests : 
- on same domain as synapse, immediatly after login I am logged out. In dev tools I see that the request on the api got a 401 unauthorized erro
- on a different domain, I still issues with the request to the admin API. Same as the previous screenshot. Maybe we should ynh_permission_create on the admin API to make it public (meh, I am not verry keen on having it open though...)

**Comment by nathanael-h** [comment]
What about something like this in installation...?
`ynh_permission_create --permission="permission" --url=${synapse_domain}/_synapse/" --auth_header=false --allowed=vistors --label="Synpase admin API"`

**Comment by tituspijean** [comment]
> What about something like this in installation...? `ynh_permission_create --permission="permission" --url=${synapse_domain}/_synapse/" --auth_header=false --allowed=vistors --label="Synpase admin API"`

That won't work, the permissions are set relatively to the main path of the app. We may run a Python command to set that one for the Synapse app though.

**Comment by collector-ynh** [comment]
I see exactly the same problem as @nathanael-h , have you found any solutions to solve this problem?

**Comment by Nalla22** [comment]
I am also experiencing the same problem, thank you for help me solve it.

**Comment by nathanael-h** [comment]
As said here, https://github.com/YunoHost-Apps/synapse_ynh/pull/301#issuecomment-1106921305 , I upgraded my synapse to the branch that could help to fix our problem. But I still have issues... Facing SSO.

**Comment by nathanael-h** [comment]
I was not using a valid endpoint for my tests. The app now works using this branch https://github.com/YunoHost-Apps/synapse_ynh/tree/add-_synapse-perm. 

If someone wants to test it: 
`sudo yunohost app upgrade synapse -u https://github.com/YunoHost-Apps/synapse_ynh/tree/add-_synapse-perm --debug`

**Comment by tituspijean** [comment]
!testme

**Comment by collector-ynh** [comment]
I have installed the synapse server from the following repository: https://github.com/YunoHost-Apps/synapse_ynh/tree/add-_synapse-perm , while attributing the necessary permissions for the "visitors" group.

ksnip_20220427-200141

The purpose of this is to make the **/_synapse** extension accessible for the **Synapse-Admin** application that I have installed from the following repository : https://github.com/YunoHost-Apps/synapse-admin_ynh/tree/fix But I am facing a new problem that is preventing me from accessing my session **Synapse-Admin**, because at the time of authentication with the user "**test**" (which has all the permissions, see image above), synapse-admin opens and closes immediately, while returning the following error message :

ksnip_20220427-193622

This is what the following command returns to me : `curl https://matrix.server.fr/_synapse/admin/v1/server_version`

ksnip_20220427-223344

I do not know how to solve this problem, because I have tested several browsers and several different networks, so the problem does not originate neither the browser nor the network. If you have a solution to solve this problem, please feel free to let me know.

**Comment by collector-ynh** [comment]
I found other people who have the same problem, with the same erroneous error message : 
https://github.com/Awesome-Technologies/synapse-admin/issues/63
https://github.com/Awesome-Technologies/synapse-admin/issues/135

They were able to identify the problem, and apparently this is because of the fact that the user does not have admin rights on the synapse server, except that I have given all possible rights and imaginable to my user on YunoHost as you can see in picture. 

What kind of privilleges do I still have to give him?

**Comment by collector-ynh** [comment]
I managed to solve the problem, by manually adding admin rights to the test user on the matrix_synapse database with the next request :

`UPDATE users SET admin = 1 WHERE name = '@test:synapse.domain.tld';`

C'est quand même étrange de devoir accorder les permission dans la BDD alors que yunohost est censé fournir tous les droits admin qu'on voit dans l'image ci-dessus. La question c'est : Est-ce que ces permissions YunoHost servent vraiment à quelque chose ?

**Comment by tituspijean** [comment]
> It's still strange to have to grant the permission in the BDD while yunohost is supposed to provide all admin rights as seen in the image above. 

⬆️ https://github.com/YunoHost-Apps/synapse_ynh#set-user-as-admin
 
> The question is: Do these YunoHost permissions really serve a purpose?

In order of appearance: obviously, yes-ish,  yes, hell yes, hell yes².

> Anyway, it's a real miracle that synapse-admin finally works, because I've been struggling to get it to work for almost four months.

It's no miracle, it's thanks to your haraswork.

**Comment by collector-ynh** [comment]
> It's no miracle, it's thanks to your hard work.

And especially thanks to your work ;)

**Comment by ericgaspar** [comment]
what do we do? Do we merge?

**Comment by lapineige** [comment]
> > If I install on the same domain as synapase
> 
> What. It is supposed to be on its own domain. How could your YunoHost let you do that?!

I can confirm it does…

**Comment by ericgaspar** [comment]
!testme

**Comment by lapineige** [comment]
This should fix #15 too.

**Comment by lapineige** [comment]
In CI:
> touch: cannot touch '/etc/nginx/conf.d/.d/synapse-admin.endpoint.conf': No such file or directory

Domain name issue ?

**Comment by ericgaspar** [comment]
!testme

**Comment by lapineige** [comment]
Is that a CI bug ?

**Comment by BobWs** [comment]
> UPDATE users SET admin = 1 WHERE name = '@test:synapse.domain.tld';

Where can I find the synapse database? 
Sorry for the noob question, but I'm new to Yunohost....

**Comment by Unfitted2434** [comment]
> > If I install on the same domain as synapase
> 
> What. It is supposed to be on its own domain. How could your YunoHost let you do that?!

It is still allowing this to happen!!!

**Comment by tituspijean** [comment]
!testme

**Pull request merged**